### PR TITLE
Bypass list feature

### PR DESCRIPTION
**Pull request by ameerhossein**
#59
I rebased the previous PR onto my own outdated main branch accidentally and caused even more conflicts. I believe this new PR is fine and can be merged correctly.

### CODE REVIEW

**Comment by markpash** [comment]
The `Use OblivionVpnService static methods...` commit is already merged I think, or maybe I'm reading it wrong.

File: app/src/main/java/org/bepass/oblivion/ConnectionState.java
Comment (markpash) [inline app/src/main/java/org/bepass/oblivion/ConnectionState.java:9]: If we instead made a `isDisconnected()` boolean method, I think the rest of the code would be simpler.

File: app/src/main/java/org/bepass/oblivion/MainActivity.java
Comment (markpash) [inline app/src/main/java/org/bepass/oblivion/MainActivity.java:50]: This code and the above code would be simpler if there was an `isDisconnected()` method.

File: app/src/main/res/layout/installed_app_item.xml
Comment (markpash) [inline app/src/main/res/layout/installed_app_item.xml:27]: If you can, squash this into the previous commit where you made this file.

**Review by markpash — APPROVED** [review]
I don't know Java so my code review skills will be poor. Let me know if my suggestions are stupid. But otherwise, this looks good. Thanks!

**Comment by markpash** [comment]
I think you deleted the wrong commit :P 
I've enabled the button in github that suggests that to update the branch. Choose the "Update with rebase" option. Hopefully it helps.

**Comment by ameerhossein** [comment]
> I think you deleted the wrong commit :P
I used the update branch button but it defaulted to merge so I had to drop it.

**Comment by markpash** [comment]
> > I think you deleted the wrong commit :P
> 
> I used the update branch button but it defaulted to merge so I had to drop it.

I as the maintainer get permissions to push to your PR branch, so I can fix some of this for you right now.
Let me know if you're ok with that.

File: app/src/main/java/org/bepass/oblivion/OblivionVpnService.java
Comment (ameerhossein) [reply app/src/main/java/org/bepass/oblivion/OblivionVpnService.java:458]: You are right. It's like this because we had two modes at first (BLACKLIST and WHITELIST) but then I realized the whitelist mode required more work from the Go core lib side so I just removed the WHITELIST branch from the conditions. I will simplify the leftovers.

File: app/src/main/res/layout/installed_app_item.xml
Comment (ameerhossein) [reply app/src/main/res/layout/installed_app_item.xml:27]: Okay

**Comment by ameerhossein** [comment]
Squashed the font commit you mentioned.

> The `Use OblivionVpnService static methods...` commit is already merged I think, or maybe I'm reading it wrong.

It's a different commit. Actually it should have been committed with the previous PR. Don't know why did it end up here. probably my mistake.

> I as the maintainer get permissions to push to your PR branch, so I can fix some of this for you right now.
Let me know if you're ok with that.

I committed your suggestions. I believe you already have permission to edit the PR if you want to change anything.

**Comment by ameerhossein** [comment]
Don't merge it yet. Android manifest needs an update again after the rebase

**Comment by ameerhossein** [comment]
Ok I squashed the android manifest commit too. Everything seems ok now in my tests.

**Review by sarobakhtiyari — APPROVED** [review]

**Pull request merged**
